### PR TITLE
Add support for tmpfs mounts under `/tmp` with hermetic tmp

### PR DESCRIPTION
**Pull request by fmeum**
This is achieved by mounting tmpfs after regular mounts in the sandbox binary as well as creating the directories at which tmpfs are mounted under the sandbox tmp directory.

### CODE REVIEW

**Comment by fmeum** [comment]
@bazel-io fork 7.1.0

File: src/test/shell/bazel/bazel_sandboxing_test.sh [hunk 1]
```diff
@@ -334,6 +334,47 @@ EOF
...
+  trap "rm -fr $tmpfs" EXIT
+  echo BAD > "$tmpfs/data.txt"
+
+  mkdir -p pkg
+  cat > pkg/BUILD <<EOF
+genrule(
+    name = "gen",
+    outs = ["gen.txt"],
+    cmd = """
+# Verify that /tmp is still hermetic and that the tmpfs under /tmp exists, but is empty.
```
Comment (lberki) [inline src/test/shell/bazel/bazel_sandboxing_test.sh:361]: nit: I'd prefer using if statements or at least double square brackets, but my preference is not strong enough to mandate it, so meh.

**Review by lberki — APPROVED** [review]

File: src/test/shell/bazel/bazel_sandboxing_test.sh (hunk 1 above)
Comment (fmeum) [reply src/test/shell/bazel/bazel_sandboxing_test.sh:361]: I am using double square brackets now.

Comment (lberki) [reply src/test/shell/bazel/bazel_sandboxing_test.sh:361]: Thanks :) Still good.

**Comment by fmeum** [comment]
@lberki Do you want me to resolve the merge conflict in the tests or are you taking care of this during the import?

**Comment by iancha1992** [comment]
@fmeum @lberki there is a conflicting file. Can you please take a look?

**Comment by fmeum** [comment]
@iancha1992 Resolved

**Comment by lberki** [comment]
@fmeum thanks for rebasing; I was about to say "I'd appreciate if you did it because today just became extraordinarily complicated for me", but you were even acausal :)

(I already approved the internal change, it's grinding its way through our submit process)

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.1.0 RC1](https://releases.bazel.build/7.1.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Pull request closed without merging**